src/scaict_uwu/service_container.py

```
@@ -0,0 +1,91 @@
+            sys.exit("CannotReplaceActiveServiceException $name")
+
+        self._service_instantiators[name] = instantiator
+
+    def create_service(self, name: str):
+        if not self.has_service(name):
+            sys.exit("NoSuchServiceException $name")
+
+        if not name in self._services_being_created:
+            sys.exit(
+                "RecursiveServiceDependencyException "
+                + "Circular dependency when creating service!"
```

Ruff (0.8.0)

Explicitly concatenated string should be implicitly concatenated

(ISC003)